3_forecast/src/define_model_tasks.R

```diff
@@ -41,95 +41,71 @@ list_tasks <- function(sites_yml='1_data/out/sites_subset.yml.ind') {
 
 # prepare a plan for downloading (from WQP) and posting (to GD) one data file
 # per state
-plan_forecasts <- function(partitions, folders) {
+plan_forecasts <- function(
+  tasks_df, folders,
+  nwis_inputs='2_munge/out/agg_nwis.rds.ind',
+  nwm_inputs='2_munge/out/agg_nwm.rds.ind',
+  remake_file
+) {
 
+  # function to sprintf a bunch of key-value (string-variableVector) pairs, then
+  # paste them together with a good separator for constructing remake recipes
   psprintf <- function(..., sep='\n      ') {
     args <- list(...)
     strs <- mapply(function(string, variables) {
-      spargs <- if(string == '') list(variables) else c(list(string), list(variables))
+      spargs <- if(string == '') list(variables) else c(list(string), as.list(variables))
       do.call(sprintf, spargs)
     }, string=names(args), variables=args)
     paste(strs, collapse=sep)
   }
 
-  # steps: partition, forecast, post, retrieve
-  partition <- scipiper::create_task_step(
-    step_name = 'partition',
+  # steps: subset, forecast, post, retrieve
+  subset <- scipiper::create_task_step(
+    step_name = 'subset',
     target_name = function(task_name, step_name, ...) {
-      sprintf('partition_%s', task_name)
+      sprintf('inputs_%s', task_name)
     },
     command = function(task_name, ...) {
+      task_info <- filter(rename(tasks_df, tn=task_name), tn==task_name)
       psprintf(
         "subset_inputs(",
-        "all_inputs=I('%s'),"=,
-        "nwm_model=I('%s'),",
-        "start_calibrate=I('%s'),",
-        "start_forecast=I('%s'))", task_name,
+        "nwis_inputs='%s',"=nwis_inputs,
+        "nwis_site=I('%s'),"=task_info$site,
```

Comment:
i was working from the world where comid_subset.tsv wasn't yet a thing, so i figured the sites_subset was the best list to start from.

now that we have two options (as soon as you make your PR, i think?), i don't feel especially strongly about one over the other. I think it'd be great to keep track of both as long as possible through the analysis so that we can always reference either without having to go look up the conversion again. to that end, i'm working on a site info table in 1_data.yml that will contain both site IDs and COMIDs.